controllers/dependencies.go

```diff
@@ -0,0 +1,106 @@
+package controllers
+
+import (
+	"context"
+
+	"os"
+
+	"path/filepath"
+
+	git "github.com/go-git/go-git/v5"
+	tekton "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	OPERATOR_PIPELINES_REPO = "https://github.com/redhat-openshift-ecosystem/operator-pipelines.git"
+	REPO_CLONE_PATH         = "/tmp/operator-pipelines/"
+	PIPELINE_MANIFESTS_PATH = "ansible/roles/operator-pipeline/templates/openshift/pipelines"
+	TASK_MANIFESTS_PATH     = "ansible/roles/operator-pipeline/templates/openshift/tasks"
+)
+
+func (r *OperatorPipelineReconciler) reconcilePipelineDependencies(meta metav1.ObjectMeta) error {
+
+	// Cloning operator-pipelines project to retrieve pipelines and tasks
+	// yaml manifests that need to be applied beforehand
+	// ref: https://github.com/redhat-openshift-ecosystem/certification-releases/blob/main/4.9/ga/ci-pipeline.md#step-6---install-the-certification-pipeline-and-dependencies-into-the-cluster
+
+	_, err := git.PlainClone(REPO_CLONE_PATH, false, &git.CloneOptions{
+		URL: OPERATOR_PIPELINES_REPO,
+	})
+	if err != nil {
+		log.Error(err, "Couldn't clone the repository for operator-pipelines")
+		return err
+	}
+	defer r.removePipelineDependencyFiles(REPO_CLONE_PATH)
```

Comment:
I think (and I might be wrong), that this method call should be place before the if statement on line 33. If cloning the repo fails in the middle, the control flow will return the error but not delete the created dir.

I overaly recommend 1) creating a tmp folder programmatically for the operator-pipelines (please check https://pkg.go.dev/os#MkdirTemp) to avoid unwanted surprises, and 2) move the defer method to line 33, after the git.PlainClone and before the `if err`

Reply:
2) I'm not sure as well if the defer function would get triggered there so just in case I moved it to line 33 as requested.
 
1) On the other hand temporary files used by programs in Linux are historically stored in the /tmp directory that happens to have, in regular systems, garbage collection in place. This is a single Linux image deployment, single process container, single user with a single purpose. It's pretty straightforward. I don't see the need to add another package/method and more complexity to address this.

Reply:
No problem.